[8.0][REF] product_extended_segmentation: Add unit tests and data demo

osvalr opened the pull request.

hbto commented on product_extended_segmentation/tests/test_extend_segmentation.py:
```diff
@@ -0,0 +1,129 @@
+        pid = False
+        if product_id._name == "product.template":
+            tmpl_id = product_id
+            pid = product.id
```
Undefined name `product`

hbto commented on product_extended_segmentation/tests/test_extend_segmentation.py:
```diff
@@ -0,0 +1,129 @@
+        self.assertTrue(before_compute != tmpl_id.standard_price)
+
+        # Production cost should be: 15 = 10(E) + 5(D) production costs
+        self.assertTrue(tmpl_id.production_cost == 15)
```
Please @osvalr provide a similar test for material_cost segmentation.

osvalr replied:
I'll change demo data `standard_price` for product D, because this value is similar to material cost when recursive computation is applied, just to get this assertion you asked and it will be easy viewed.

**UPDATE**: It's not required, not doing it

osvalr replied:
done

hbto commented on product_extended_segmentation/tests/test_extend_segmentation.py:
```diff
@@ -0,0 +1,129 @@
+        self.compute_price_from_wizard(self.product_e_id, True, True)
+
+        # The cost for D-child have changed
+        self.assertTrue(d_before_compute != tmpl_d_id.standard_price)
```
@osvalr Though you have explicitly assessed that STD has changed,
Assess too, the segmentation, providing which were the expected values to be assessed.
What were the values expected for M, L, P &S, and asserting against received values.

Commit:
[REF] product_extended_segmentation

- add demo, add unit test for D product-related bom routing cost computation
- move workcenter some lines up for organize demo data in order to locate it, change cost for product E, add more testing for use cases
- add material value challenge for 3rd use test case
- detail cost values for D and E products en recursive calls
- add increasing cost recursive computation test case modifying starting cost for BoM product inherited

hbto commented on product_extended_segmentation/demo/demo.xml:
```diff
@@ -0,0 +1,186 @@
+        <record id="rev_production_analytic_account" model="account.analytic.account">
+            <field name="name">PRODUCTION_ANALYTIC_ACCOUNT</field>
+            <field ref="account.income_view" name="parent_id"/>
+        </record>
```
@osvalr Get rid of these recorded above. They not used within test.

osvalr replied:
done

Commit:
[REF] product_extended_segmentation: remove unused demo data
